### PR TITLE
fishplugins: forgit fix dependency linking

### PR DESCRIPTION
**Pull request by happysalada**
###### Motivation for this change

dependency linking is currently broken. The plugin depends on fzf and git

I've tested building and I've briefly checked that the resulting file doesn't seem broken, but I'm not sure if that is the right way to proceed.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [x] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 120962` at 7c861b45 run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built successfully:</summary>
  <ul>
    <li>fishPlugins.forgit</li>
  </ul>
</details>
<details>
  <summary>3 suggestions:</summary>
  <ul>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/missing-phase-hooks.md">missing-phase-hooks</a>

`buildPhase` should probably contain `runHook preBuild` and `runHook postBuild`.

Near pkgs/shells/fish/plugins/build-fish-plugin.nix:37:10:
```
   |
37 |   inherit unpackPhase configurePhase buildPhase;
   |          ^
```
</li>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/missing-phase-hooks.md">missing-phase-hooks</a>

`checkPhase` should probably contain `runHook preCheck` and `runHook postCheck`.

Near pkgs/shells/fish/plugins/build-fish-plugin.nix:69:3:
```
   |
69 |   checkPhase = ''
   |   ^
```
</li>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/missing-phase-hooks.md">missing-phase-hooks</a>

`configurePhase` should probably contain `runHook preConfigure` and `runHook postConfigure`.

Near pkgs/shells/fish/plugins/build-fish-plugin.nix:37:10:
```
   |
37 |   inherit unpackPhase configurePhase buildPhase;
   |          ^
```
</li>
  </ul>
</details>

**Comment by legendofmiracles** [comment]
Before i give my thumbs up on this, how can i test this? 
Adding it to the home-manager option `programs.fish.plugins` doesn't work, as it only expects a set.

**Comment by happysalada** [comment]
yes, I was surprised, but just adding it to your packages work. so if you are using home-manager, just your regulard `home.packages`

**Comment by legendofmiracles** [comment]
It also works without this patch 🤔

**Comment by pacien** [comment]
@legendofmiracles wrote:
> It also works without this patch

Are you sure that you don't already have `fzf` and `git` in your usual `PATH`?
This could be hiding the issue.

File: pkgs/shells/fish/plugins/forgit.nix [hunk 1]
```diff
@@ -4,7 +4,13 @@ buildFishPlugin rec {
...
   version = "unstable-2021-04-09";
 
-  buildInputs = [ git fzf ];
+  preFixup = ''
+    substituteInPlace $out/share/fish/vendor_conf.d/forgit.plugin.fish \
+      --replace "fzf " "${fzf}/bin/fzf "
+
+    substituteInPlace $out/share/fish/vendor_conf.d/forgit.plugin.fish \
+      --replace "git " "${git}/bin/git "
+  '';
```
Comment (pacien) [inline pkgs/shells/fish/plugins/forgit.nix:13]: You can combine both:

```suggestion
  preFixup = ''
    substituteInPlace "$out/share/fish/vendor_conf.d/forgit.plugin.fish" \
      --replace "fzf " "${fzf}/bin/fzf " \
      --replace "git " "${git}/bin/git "
  '';
```

File: pkgs/shells/fish/plugins/default.nix [hunk 2]
```diff
@@ -15,7 +15,7 @@ lib.makeScope newScope (self: with self; {
 
   foreign-env = callPackage ./foreign-env { };
 
-  forgit-fish = callPackage ./forgit.nix { };
+  forgit = callPackage ./forgit.nix { };
```
Comment (pacien) [inline pkgs/shells/fish/plugins/default.nix:18]: Nitpicks:
* it would be better to put the attribute rename in a separate commit,
* commit format should also be of the form: "fishPlugins.forgit: ...".

**Comment by happysalada** [comment]
@legendofmiracles you probably have fzf in your path. If you try to remove it, it should not work with the old version. (removing git should produce the same)
